cdn/supervisor/cdn/cache_writer.go

```diff
@@ -139,12 +144,15 @@ loop:
 				break loop
 			}
 			backSourceLength += n
-
+			end := time.Now()
 			jobCh <- &piece{
-				taskID:       seedTask.ID,
-				pieceNum:     uint32(curPieceNum),
-				pieceSize:    uint32(seedTask.PieceSize),
-				pieceContent: bb,
+				taskID:            seedTask.ID,
+				pieceNum:          uint32(curPieceNum),
+				pieceSize:         uint32(seedTask.PieceSize),
+				pieceContent:      bb,
+				downloadCost:      uint64(end.Sub(start).Milliseconds()),
+				beginDownloadTime: uint64(start.UnixMilli()),
```

Comment:
Use time.Now().UnixNano(), refer to https://github.com/dragonflyoss/Dragonfly2/blob/main/client/daemon/peer/piece_manager.go#L186